cmake/OpenCVDetectInferenceEngine.cmake

```diff
@@ -25,7 +25,7 @@ macro(ocv_ie_find_extra_libraries find_prefix find_suffix)
     string(REPLACE "${find_prefix}" "" library "${library}")
     if(library STREQUAL "inference_engine" OR library STREQUAL "inference_engined")
       # skip
-    else()
+    elseif(NOT TARGET ${library})
```

Comment:
This doesn't look like a robust solution.

We could limit DLDT files suffixes near line 65:
```
if(NOT DEFINED OPENCV_IE_LIBRARY_SUFFIXES)
  set(OPENCV_IE_LIBRARY_SUFFIXES "${CMAKE_FIND_LIBRARY_SUFFIXES}")
endif()
... replace CMAKE_FIND_LIBRARY_SUFFIXES => OPENCV_IE_LIBRARY_SUFFIXES below ..
```

and specify `OPENCV_IE_LIBRARY_SUFFIXES=.so` through CMake.

---

BTW, It is better to move DLDT testing stuff into a separate directory or provide `find_package(InferenceEngine)` for build directory.

Reply:
1. Removed the condition for target. It turned our that both version of library (.so and .a) are just old artifacts after my experiments with static libraries.
2. We have `InferenceEngineConfig.cmake` for build tree as well, but build flow is organized in such manner that IE and OpenCV are build on different machines and hardcoded paths inside the file with targets don't work.